js/viewModels/subjectViewModel.js

```diff
@@ -0,0 +1,198 @@
+class SubjectViewModel {
+  #subjectList;
+  #taskViewModel;
+
+  constructor(taskViewModel) {
+    this.#subjectList = new Map();
+    STATE_LIST.forEach((state) => this.#subjectList.set(state, [])); // Init all columns
+    this.#taskViewModel = taskViewModel;
+  }
+
+  /**
+   * Adds a new subject to the specified state column and updates the view.
+   *
+   * @param {string} params.title - The title of the new subject.
+   * @param {string} [params.state=OPEN] - The state column where the subject will be added. Default value is `OPEN`.
+   *
+   * @returns {Subject} The newly created subject.
+   */
+  addSubject({ title, state = OPEN }) {
+    const newSubject = new Subject({ title, state });
+
+    this.#subjectList
+      .get(state) // Get the column where the subject will be inserted
+      .push(newSubject);
+
+    this.render();
+    return newSubject;
+  }
+
+  /**
+   * Removes a subject from the specified state column and updates the view.
+   *
+   * @param {string} params.targetId - The ID of the subject to be removed.
+   * @param {string} params.state - The state column from which the subject will be removed.
+   */
+  deleteSubject({ targetId, state }) {
+    // Get the column from which the subject will be removed
+    const subjectList = this.#subjectList.get(state);
+
+    // Get the index of the target subject by its ID
+    const targetIndex = subjectList.findIndex(
+      (subject) => subject.getId() === targetId
+    );
+    // Remove the target subject from the column
+    subjectList.splice(targetIndex, 1);
+
+    this.render();
+  }
+
```

Comment:
```suggestion
  deleteSubject({ targetId, state }) {
    // Get the column from which the subject will be removed
    const subjectList = this.#subjectList.get(state);

    // Get the index of the target subject by its ID
    const targetIndex = subjectList.findIndex(
      (subject) => subject.getId() === targetId
    );
    // Remove the target subject from the column
    subjectList.splice(targetIndex, 1);

if(targetIndex !== -1) {
subjectList.splice(targetIndex, 1);
}
    this.render();
  }

```

요렇게 예외처리를 추가해주는 것도 좋을 것 같아용

Reply:
제가 예외처리를 많이 빼먹었네요!ㅋㅋㅋ 감사합니다🫶🫶🫶